docs/source/datastore.rst

```diff
@@ -130,3 +130,80 @@ related syntax.
         }
     }
 
+Securing secrets in key value store (admin only)
+------------------------------------------------
+
+.. note::
+
+    This guide and the corresponding implementation is alpha quality. We are working on the feature
+    and feedback is welcome. Unless the feature matures and deployment issues identified and fixed,
+    no guarantee is made w.r.t ``security`` of the credentials stored in key value store.
```

Comment:
I love it when we come with a security feature and then say beware dragon be living here :P